database/postgres-views/README.md

```diff
@@ -0,0 +1,266 @@
+
+To create a `materialized view` -> **4097.513 ms**.
+
+```sql
+CREATE MATERIALIZED VIEW copr_target_builds_for_pull_requests AS
+    SELECT git_projects.project_url, git_projects.repo_name, copr_build_targets.build_id, pipelines.datetime
+    FROM git_projects
+    JOIN pull_requests ON git_projects.id = pull_requests.project_id
+    JOIN job_triggers ON job_triggers.trigger_id = pull_requests.id
+    JOIN pipelines ON pipelines.job_trigger_id = job_triggers.id
+    JOIN copr_build_groups ON copr_build_groups.id = pipelines.copr_build_group_id
+    JOIN copr_build_targets ON copr_build_targets.copr_build_group_id = copr_build_groups.id
+    ORDER BY git_projects.project_url, git_projects.instance_url;
+```
+
+---
+
+To refresh the `materialized view` -> **5526.618 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+To query the `materialized view` -> **904.686 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+To create a `normal view` -> **25.266 ms**
+
+```sql
+CREATE VIEW copr_target_builds_for_pull_requests AS
+    SELECT git_projects.project_url, git_projects.repo_name, copr_build_targets.build_id, pipelines.datetime
+    FROM git_projects
+    JOIN pull_requests ON git_projects.id = pull_requests.project_id
+    JOIN job_triggers ON job_triggers.trigger_id = pull_requests.id
+    JOIN pipelines ON pipelines.job_trigger_id = job_triggers.id
+    JOIN copr_build_groups ON copr_build_groups.id = pipelines.copr_build_group_id
+    JOIN copr_build_targets ON copr_build_targets.copr_build_group_id = copr_build_groups.id
+    ORDER BY git_projects.project_url, git_projects.instance_url;
+```
+
+---
+
+To query the `normal view` -> **2616.131 ms**
+
+```
+SELECT * FROM copr_target_builds_for_pull_requests;
+```
+
+---
+
+Since we have to refresh a `materialized view`, like this, every time we use it (the data it contains change too often) sadly **there is no improvement**.
+
+# Normal views
+
+Normal views, _once created, are easy to work with_ but I can not really say that they save us time.
+
+To get the usage statistics page from the service api (https://service.localhost/api/) it tooks me **4 seconds**
+
+```logs
+service             | [Fri May 19 12:57:34.145012 2023] [wsgi:error] [pid 6:tid 39] [remote 10.89.0.4:39636] 2023-05-19 12:57:34,144 INFO sqlalchemy.engine.Engine [cached since 1.482e+04s ago] {'type_1': 'issue'}
+service             | 10.89.0.4 - - [19/May/2023:12:57:30 +0000] "GET /api/usage HTTP/1.1" 200 120160
+```
+
+To get all the data we need with a single query (saved as a view named `jobs_count`) it tooks me **3277.724 ms**.
+
+We can move the complexity below at sql level and simplify our code, this is all.
+
+The views are not really supported in sqlalchemy, there are two projects that implement them somehow (`sqlalchemy-utils` and `sqlalchemy-views`) but this are "jung projects" I would not count on them for the `packit-service` code.
```

Comment:
`junk` or `young` projects?..;)

Since we are making this public, we might want to use more polite wording I would say... :smile: